doc/src/devdocs/pkgimg.md

```diff
@@ -0,0 +1,47 @@
+# Package Images
+
+Julia package images provide object (native code) caches for Julia packages.
+They are similar to Julia's [system image](@) and support many of the same features.
+In fact the underlying serialization format is the same, and the system image is the base image that the package images are build against.
+
+## High-level overview
+
+Package images are shared libraries that contain both code and data. Like `.ji` cache files they are generated per package. The data section contains both global data (global variables in the package) as well as the necessary metadata about what methods and types are defined by the package. The code section contains native objects that cache the final output of Julia's LLVM based compiler.
+
+The command line option `--pkgimage-native-code=no` can be used to turn-off object caching for this session. Note that this means that cache files have to be regenerated.
+
+!!! note
+    While the package images present themselves as native shared libraries they are only an approximation thereof. You will not be able to link against them from a native program and they must be loaded from Julia.
+
+
+## Linking
+
+Since the package images partially contain native code, we must run a linker over them before we can use them. You can use the environment variable `JULIA_VERBOSE_LINKING` to make the pkgimage linking process verbose.
```

Comment:
```suggestion
Since the package images contain native code, we must run a linker over them before we can use them. You can set the environment variable `JULIA_VERBOSE_LINKING` to `true` to make the package image linking process verbose.
```
I think "partially" is superfluous in this context, no? 

Further, I'd either give complete information on what to do with the env var or link to the section where it is defined.